### PR TITLE
[FrontDoor] az network front-door frontend-endpoint enable-https: allow secret-version to be optional to use Latest version

### PR DESCRIPTION
**Pull request by t-bzhan**
This PR will fix https://github.com/Azure/azure-cli-extensions/issues/3151
---

This checklist is used to make sure that common guidelines for a pull request are followed.

### General Guidelines

- [x] Have you run `azdev style <YOUR_EXT>` locally? (`pip install azdev` required)
- [x] Have you run `python scripts/ci/test_index.py -q` locally?

For new extensions:

- [ ] My extension description/summary conforms to the [Extension Summary Guidelines](https://github.com/Azure/azure-cli/blob/dev/doc/extensions/extension_summary_guidelines.md).

### About Extension Publish

There is a pipeline to automatically build, upload and publish extension wheels.  
Once your PR is merged into master branch, a new PR will be created to update `src/index.json` automatically.  
The precondition is to put your code inside this repo and upgrade the version in the PR but do not modify `src/index.json`.

### CODE REVIEW

**Comment by yonzhan** [comment]
FrontDoor

File: src/front-door/azext_front_door/_help.py [hunk 1]
```diff
@@ -148,6 +148,14 @@
     type: command
     short-summary: Disable HTTPS protocol for a custom domain.
 """
+
+helps['network front-door frontend-endpoint wait'] = """
```
Comment (kairu-ms) [inline src/front-door/azext_front_door/_help.py:152]: Please add test for this command

Comment (t-bzhan) [reply src/front-door/azext_front_door/_help.py:152]: Thanks, added.

File: src/front-door/azext_front_door/_params.py [hunk 2]
```diff
@@ -65,7 +65,8 @@ def load_arguments(self, _):
     with self.argument_context('network front-door', arg_group='HTTPS') as c:
         c.argument('certificate_source', arg_type=get_enum_type(FrontDoorCertificateSource), help='Certificate source to enable HTTPS.')
         c.argument('secret_name', help='The name of the Key Vault secret representing the full certificate PFX')
-        c.argument('secret_version', help='The version of the Key Vault secret representing the full certificate PFX')
+        c.argument('secret_version', help='The version of the Key Vault secret representing the full certificate PFX, '
```
Comment (jsntcy) [inline src/front-door/azext_front_door/_params.py:68]: Are there any other versions supported besides `Latest`?

Comment (t-bzhan) [reply src/front-door/azext_front_door/_params.py:68]: Yes, I've added two test: One for Latest version and one for specific version.

For the time being, those tests will fail as our existing enable-https implementation has ignored the LROPoller which will cause failures in TearDown, I've marked them as skipped.

Another way to work around this is we could wait for the operations to finish using az network frontend-endpoint wait, but that only works in live_only model, it seems that VCR.py does not work well with the ignored underlying LRO poll, and each test will take more than 1 hour to finish.

**Review by jsntcy — APPROVED** [review]
:shipit:

**Pull request merged**